spirv_glsl.cpp

```diff
@@ -1461,6 +1461,8 @@ string CompilerGLSL::layout_for_variable(const SPIRVariable &var)
 
 	// GL 3.0/GLSL 1.30 is not considered legacy, but it doesn't have UBOs ...
 	bool can_use_buffer_blocks = (options.es && options.version >= 300) || (!options.es && options.version >= 140);
+        // pretend no UBOs when options say so
+	can_use_buffer_blocks &= options.enable_buffer_blocks;
```

Comment:
Use logical operations here. This hidden bool -> arithmetic and -> bool will trigger warnings on many compilers.